supplementary/alica_ros1/alica_tests/src/test/test_config_change.cpp

```diff
@@ -43,11 +43,11 @@ TEST_F(AlicaNotInitialized, TestUpdatingComponents_002)
 TEST_F(AlicaNotInitialized, TestUpdatingComponents_003)
 {
     EXPECT_TRUE(ac->setOption<bool>("Alica.AllowIdling", false));
-    EXPECT_FALSE(PartialAssignment::isIdlingAllowed());
+    EXPECT_TRUE(PartialAssignment::isIdlingAllowed());
```

Comment:
why did these have to change?

Reply:
I have tested a lot of what is happening in devel and this test is wrong.

In order for it to succeed, `PartialAssignment::isIdlingAllowed()` has to be false, and this is impossible without `ac->init` being called. So I made some tests and saw that somehow ac->init from the upper test is leaking stuff to this test. When I isolated this test and executed just this `AlicaNotInitialized.TestUpdatingComponents_003` it failed in devel. Also, I made an even worse test case, I commented out `ac->init` from every test case but the first in this file and everything worked, so it is definitely wrong.

In my branch everything is properly terminated when going from one test case to the other.

It's good to have a call with you to explain everything I discovered and also for you to explain me more stuff. But pretty sure this test right now in devel is wrong.

Reply:
The test may be wrong but it does not make sense to set Alica..AllowIdling as false and then expect isIdlingAllowed to be true, so the test would need additional changes to make sense

Reply:
@Dale-Koenig actually the test itself mentions that this configuration only takes effect after the engine is initialised. In the first check, the engine is not initialised & setOption has not effect. Then the test goes on the call ac->init() after which setOption takes effect. 

So, as per the code isIdlingAllowed() should return true which is correct. The question is why this test was passing till now. Maybe because this method is returning a static variable there is some leakage between tests, but I'm not sure, because I couldn't find another place where this variable is being set explicitly

Reply:
Hmm okay, then the test is weird but it is fine now, and was passing before due to the leaks mentioned from previous tests and static variables

Reply:
@rogeriofonteles even if you ensure everything is properly terminated between tests, since this is a static variable, things can still get messed up I think. The ideal solution would be to make this a non static variable

Reply:
Fine to fix in a follow up imo